packages/address/src/utils.ts

```diff
@@ -103,3 +104,27 @@ export const addressify = (addressLike: AddressLike | ContractIdLike): AbstractA
 };
 
 export const getRandomB256 = () => hexlify(randomBytes(32));
+
+/**
+ * An EvmAddress contains a B256 with the first 12 bytes cleared, this function
+ * takes a b256 and returns itself with the first 12 bytes cleared
+ *
+ * @param b256 - the address to clear
+ * @returns b256 with first 12 bytes cleared
+ */
+export const getEvmB256fromB256 = (b256: B256Address): B256AddressEvm => {
```

Comment:
Would moving this utility inside the `address` class make sense?

Similar to how `toEvmAddress` was added, this could be moved as `fromEvmAddress`.

Reply:
So I did this because it seems to follow the currently implemented pattern of the other `Address` class methods that call utility functions. We seem to abstract away most of the actual address conversion to the utility functions, with `Address` handling the shape.

This was probably done to adhere to DRY hence the `AbstractAddress` and the utility functions, however i think that abstraction is a bit redundant. So we probably could refactor all of that logic to `Address`, however then it is starting to become a pretty bloated class.

I _could_ move this logic out of utils if preferred but to me it seems to follow the current implementation.

Reply:
Thank's for the detailed answer.

My concern is less with the placement of the code block and more with the ease of use.

For example, if one wants to instantiate an `Address` from an `EvmAddress` (I suppose this is a valid use case), it can be cumbersome to figure out (or to explain) that using the `getEvmB256fromB256` utility is required.

What if we renamed it to something like `clearFirst12BytesFromB256` (to be explicit), kept it in the utils land, and then called it from a new `fromEvmAddress` static method defined in the `Address` class?

Reply:
Makes sense 👍🏻